incubator/kafka/templates/kafka-petset.yaml

```diff
@@ -0,0 +1,94 @@
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: "{{ printf "%s-%s-broker" .Release.Name .Values.Name | trunc 24 }}"
+  labels:
+    heritage: {{.Release.Service | quote }}
+    release: {{.Release.Name | quote }}
+    chart: "{{.Chart.Name}}-{{.Chart.Version}}"
+    component: "{{.Release.Name}}-{{.Values.Component}}"
+  annotations:
+    "helm.sh/created": {{.Release.Time.Seconds | quote }}
+    service.alpha.kubernetes.io/tolerate-unready-endpoints: "true"
+spec:
+  ports:
+  - port: 9092
+  clusterIP: None
+  selector:
+    component: "{{.Release.Name}}-{{.Values.Component}}"
+    app: kafka
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: "{{ printf "%s-%s" .Release.Name .Values.Name | trunc 24 }}"
+  labels:
+    heritage: {{.Release.Service | quote }}
+    release: {{.Release.Name | quote }}
+    chart: "{{.Chart.Name}}-{{.Chart.Version}}"
+    component: "{{.Release.Name}}-{{.Values.Component}}"
+  annotations:
+    "helm.sh/created": {{.Release.Time.Seconds | quote }}
+spec:
+  ports:
+  - port: 9092
+  selector:
+    component: "{{.Release.Name}}-{{.Values.Component}}"
+    app: kafka
+---
+apiVersion: apps/v1alpha1
+kind: PetSet
+metadata:
+  name: "{{ printf "%s-%s" .Release.Name .Values.Name | trunc 24 }}"
+  labels:
+    app: kafka
+    heritage: {{.Release.Service | quote }}
+    release: {{.Release.Name | quote }}
+    chart: "{{.Chart.Name}}-{{.Chart.Version}}"
+    component: "{{.Release.Name}}-{{.Values.Component}}"
+  annotations:
+    "helm.sh/created": {{.Release.Time.Seconds | quote }}
+spec:
+  serviceName: "{{ printf "%s-%s" .Release.Name .Values.Name | trunc 24 }}"
+  replicas: {{ default 3 .Values.Replicas }}
+  template:
+    metadata:
+      labels:
+        component: "{{ printf "%s-%s" .Release.Name .Values.Name | trunc 24 }}"
+      annotations:
+        pod.alpha.kubernetes.io/initialized: "true"
+        pod.alpha.kubernetes.io/init-containers: '[
+        ]'
+    spec:
```

Comment:
I could just add the same test for readiness as well. Not sure how valuable that would be though, in Kafka's situation.